kcidb/monitor/output.py

```diff
@@ -176,16 +179,18 @@ def render(self):
         subject = TEMPLATE_ENV.from_string(self.message.subject).render(ctx)
         subject_extra_bytes = \
             len(subject.encode()) - NOTIFICATION_MESSAGE_SUBJECT_MAX_LEN
-        assert subject_extra_bytes <= 0, \
-            f"Subject is {subject_extra_bytes} bytes too long"
+        if subject_extra_bytes > 0:
+            subject = subject[:NOTIFICATION_MESSAGE_SUBJECT_MAX_LEN]
+            LOGGER.warning("Subject is %s bytes too long", subject_extra_bytes)
         assert NOTIFICATION_MESSAGE_SUBJECT_RE.fullmatch(subject), \
             f"Subject is invalid, must match " \
             f"{NOTIFICATION_MESSAGE_SUBJECT_RE.pattern} regex"
         body = TEMPLATE_ENV.from_string(self.message.body).render(ctx)
         body_extra_bytes = \
             len(body.encode()) - NOTIFICATION_MESSAGE_BODY_MAX_LEN
-        assert body_extra_bytes <= 0, \
-            f"Body is {body_extra_bytes} bytes too long"
+        if body_extra_bytes > 0:
+            body = body[:NOTIFICATION_MESSAGE_BODY_MAX_LEN]
+            LOGGER.warning("Body is %s bytes too long", body_extra_bytes)
```

Comment:
You need to from limiting by bytes to limiting by characters before you can meaningfully truncate messages. See how those differ e.g. in https://en.wikipedia.org/wiki/UTF-8#Encoding